54 mantainer guide (#71)
CoronaWhy/task-geo: docs/maintainer_guide.rst

Maintainer Guide
================

> This document purpose is to define and concrete the tasks related to maintaining the github repository

Reviewer: To "concrete" doesn't exist afaik. Maybe "to concretely define", or "to define and make concrete".

   albeit unexpected behavior, the tag ``wontfix`` can be added and the issue closed.

2. **Assignement:** After we create the branch, we create a Trello card linking it to the github issue
   and explaining the problem, the project manager will find someone to fix the bug.

3. **Development:** After the project manager finds somebody to fix the bug, this is the stage where the
   bug is actually fixed.
4. **Resolution:**. After the bug has been fixed, the changes are submited with a Pull Request.

Feature requests
****************

Feature requests are request for improvements in our package.

They should include two parts:

- **Motivation:** Either something that is not correct and can be improved, or a new feature they
  consider the package should provide.

- **Proposal of solution:** How they propose the issue to be solved, or how, in general terms, they
  whish the behavior was after everything is solved.

Feature request workflow
........................

After a feature request is created, it passes through 4 steps:

1. **Verification and design:** After receiving the feature request, we must make sure that the proposal:
   

>    a) Makes sense in the scope of this project and his motivations.

Reviewer 2: ```suggestion
   a) Is within the scope of this project and its motivations.
```

   consideration:
   
   a) What structures will be required to be created?
   b) Which inputs, outputs and functionality will have each part?
   c) How will it be used by the final user?
   d) Are components in our codebase with similar functionality to any of the components that
      can be used, or modified to fulfill the requirements?
   e) In which part of the repository should be added this components?
   f) There will be any potential conflicts with other components?

   We will add the design proposal in a comment.

2. **Assignement:** After we create the branch, we create a Trello card linking it to the github
   issue and a brief summary of the design proposal, the project manager will find someone to
   implement it.

3. **Development:** After the project manager finds somebody to implement the feature, this is
   the stage where coding happens.

4. **Resolution:**. After the feature has been, the changes are submited with a Pull Request.

Support
*******

Support issues came from users who tried to use our package but the the documentation ( or the
lack of thereof) wasn't helpful enough to get the results they wanted.

> Usually this issues don't require developing code, just explaining the user how to use our package.

Reviewer 2: ```suggestion
Usually to resolve these issues don't require developing code, just explaining to the user how to use our package.
```

In the case that documentation is missing or incomplete, a Pull Request can be open to improve the
documentation.

Tasks
*****

Tasks are related to issues created from Trello and contain improvements to be done discussed
in the trello card or for which general documentation already exists, so no further work for the
maintainer is required.

Pull requests
-------------

> Pull Requests are open by collaborators who whish their work to be included in the repository.

Reviewer 2: ```suggestion
Pull Requests are opened by collaborators who wish their work to be included in the repository.
```

   the corresponding changes.

2. **Continous Integration / Build**: Github will automatically launch a build job for every new
   Pull Request that is open, that will run the tests and linters. If the build fails, we will add
   the ``ci-fail`` label and ask the submitter to fix it. If the build passes, we add the ``ci-ok``
   label and move to the next step.

3. **Output / notebooks / examples / documentation format:** Now we will review that the outputs 
   and the components behavior is as expected, usually this can be done with:

   - Cloning the fork/ checking out the branch
   - Create a new virtualenv and install from scratch the module
   - Run the notebooks provided, if any, seeing if the outputs change with the ones submitted.
   - Generate the docs and review the docs added for the submission.
   - Use the public API and see that the output is compliant with the specifications.

   If we found any issues, we will add the ``output-fail`` label and ask for the changes. If 
   everything is ok, we add the ``output-ok`` label and continue to the next step.

4. **Peer review / Code review:** This is the final step, where we read the code and analyze it 

> looking for flaws, things that should be looked include:

Reviewer 2: ```suggestion
for flaws. Things that should be considered include:
```
